xwiki-commons-core/xwiki-commons-observation/xwiki-commons-observation-local/src/main/java/org/xwiki/observation/internal/DefaultObservationManager.java

```diff
@@ -221,6 +221,10 @@ public void addListener(EventListener eventListener)
     @Override
     public void removeListener(String listenerName)
     {
+        // this might happen in some tests
+        if (this.listenersByEvent == null) {
+            initializeListeners();
+        }
```

Comment:
hmm this doesn't look very nice.... First the comment is not very appealing and then initializing listeners to remove one feels weird :)

Reply:
PS: Small detail: Would be great if you could start your sentences with an uppercase in comments :)

Reply:
It's also not good to change runtime code for tests. The justification is not good enough IMO.

Reply:
I'll encapsulate the entire method in a `if (this.listenersByEvent != null)` to avoid the NPE. But I'll also propose a small change in the test since basically it proves that it calls the remove even if there's no listeners yet.

Reply:
> it proves that it calls the remove even if there's no listeners yet.

That's not a bad test actually. It's the implementation that seems wrong. Why remove a listener if there are none? I think that's what you meant by surrounding the whole method with an `if`.